lib/licensed/sources/nuget.rb

```diff
@@ -0,0 +1,226 @@
+          return @license if defined? @license
+          return @license = package_file.license if package_file && package_file.license
+          return @license = nuspec_local_license_file.license if nuspec_local_license_file && nuspec_local_license_file.license
+          return @license = nuspec_remote_license_file.license if nuspec_remote_license_file && nuspec_remote_license_file.license
+          super
+        end
+
+        def package_name
+          return unless @metadata
+          @metadata["name"]
+        end
+
+        def package_file
+          @package_file ||= NuspecFile.new(nuspec_contents, nuspec_path)
+        end
+
+        def project_files
+          @project_files ||= [license_files, readme_file, package_file, nuspec_local_license_file, nuspec_remote_license_file].flatten.compact
+        end
+
+        def nuspec_path
+          name = @metadata["name"]
+          File.join(self.path, "#{name}.nuspec")
+        end
+
+        def nuspec_contents
+          return unless nuspec_path
+          @nuspec_contents ||= File.read(nuspec_path)
+        end
+
+        # Look for a <license type="file"> element in the nuspec that points to an
+        # on-disk license file (which licensee may not find due to a non-standard filename)
+        def nuspec_local_license_file
+          return @nuspec_local_license_file if defined?(@nuspec_local_license_file)
+          return unless nuspec_contents
+
+          match = @nuspec_contents.match LICENSE_FILE_REGEX
+          if match && match[1]
+            license_path = File.join(File.dirname(nuspec_path), match[1])
+            return unless File.exist?(license_path)
+            license_data = File.read(license_path)
+            @nuspec_local_license_file = Licensee::ProjectFiles::LicenseFile.new(license_data, license_path)
+          end
+        end
+
+        # Look for a <licenseUrl> element in the nuspec that either is known to contain a license identifier
+        # in the URL, or points to license text on the internet that can be downloaded.
+        def nuspec_remote_license_file
+          return @nuspec_remote_license_file if defined?(@nuspec_remote_license_file)
+          return unless nuspec_contents
+
+          match = nuspec_contents.match LICENSE_URL_REGEX
+          if match && match[1]
+            url = original_url = match[1]
+
+            # Skip downloading if the URL contains a license identifier itself
+            if known_license = inspect_url_for_license(url)
+              return @nuspec_remote_license_file = KnownLicenseFile.new(known_license, { uri: url })
+            end
+
+            unless ignored_url?(url)
+              # Transform URLs that are known to return HTML but have a corresponding text-based URL
+              url = get_text_content_url(url)
+
+              # Attempt to fetch the license content
+              license_data = retrieve_license(url)
```

Comment:
There’s still a good number of licenses that will come back as HTML. Should these be ignored? Included as HTML? Parsed to text?

Reply:
It looks like licensee already handles [html files](https://github.com/licensee/licensee/blob/4a9133ed3abf46fbe5a30ca1f1846484807f6eb2/lib/licensee/content_helper.rb#L295-L301) though you might need to jump through some hoops to get that working with the license data downloaded from a url.

Alternatively, since [reverse_markdown](https://github.com/xijo/reverse_markdown) is already installed transitively through licensee it may make sense to add it to `licensed.gemspec` directly and call it manually to make it easier.

Either way, as long as we can give licensee the best chance to parse and classify the content, 👍

Reply:
Added reverse_markdown, appears to work better than I expected!